gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/ProjectConverter.java

```diff
@@ -14,36 +14,28 @@
  * limitations under the License.
  */
 
-package com.google.android.gnd.persistence.remote.firestore;
+package com.google.android.gnd.persistence.remote.firestore.schema;
 
 import static com.google.android.gnd.util.Localization.getLocalizedMessage;
 
-import androidx.annotation.Nullable;
 import com.google.android.gnd.model.Project;
+import com.google.android.gnd.persistence.remote.DataStoreException;
 import com.google.firebase.firestore.DocumentSnapshot;
-import com.google.firebase.firestore.IgnoreExtraProperties;
-import java.util.Map;
 import java8.util.Maps;
 
-@IgnoreExtraProperties
-public class ProjectDoc {
-  @Nullable public Map<String, String> title;
+/** Converts between Firestore documents and {@link Project} instances. */
+class ProjectConverter {
 
-  @Nullable public Map<String, String> description;
-
-  // TODO: Add AuditInfoDoc for created and lastModified.
-
-  @Nullable public Map<String, LayerDoc> featureTypes;
-
-  public static Project toObject(DocumentSnapshot doc) {
-    ProjectDoc pd = doc.toObject(ProjectDoc.class);
+  static Project toProject(DocumentSnapshot doc) throws DataStoreException {
+    ProjectDocument pd = doc.toObject(ProjectDocument.class);
```

Comment:
Instead of doing a conversion here, could we just add a single layer of indirection to the DocSnapshots? -- so that their first interior/nested field is the representation of a Project, instead of having them represent the project itself? --or would that be rather fussy/pointless?

Reply:
Based on:

> Because they come from the enclosing map, e.g.
> 
> ```
> {
>    'id1234': {
>      // nested object
>    }
> }
> ```

Seems like it'd be a pain to do what I'm suggesting/asking here

Reply:
Firebase client libraries return you `DocumentSnapshot`s; we'd still need to ask the API to convert them to objects for us someplace.

Reply:
If we wanted to abstract his I suppose we could define the classes to convert to declaratively in some sort of a registry, then define a standard interface for converters, but that would lead to a of boilerplate, and not allow us to provide custom args to converter methods as we do now, forcing us to propose the args to member variables of hypothetical converter instances. Oy. 🙄 Open to other ideas if you had something else in mind though!